src/components/PopupWithForm.js

```diff
@@ -21,20 +22,41 @@ export default class PopupWithForm extends Popup {
     return inputItems;
   }
 
+  _getButtonText() {
+    return this._submitButton.textContent;
+  }
+
+  _setButtonText(text) {
+    this._submitButton.textContent = text;
+  }
+
+  renderLoading(isLoading) {
+    if (isLoading) {
+      this._setButtonText("Saving...");
+      this._submitButton.disabled = true;
+    } else {
+      this._setButtonText(this._originalButtonText);
+      this._submitButton.disabled = false;
+    }
+  }
+
   _handleSubmit(event) {
     event.preventDefault();
     const inputValues = this._getInputValues();
+    this.renderLoading(true);
     this._handleFormSubmit(inputValues);
-    this._resetForm();
-    this.close();
   }
 
   _setEventListeners() {
     super.setEventListeners();
     this._popupForm.addEventListener("submit", (e) => this._handleSubmit(e));
   }
 
-  _resetForm() {
+  resetForm() {
     this._popupForm.reset();
   }
+
+  close() {
+    super.close();
```

Comment:
You shouldn’t create the **same methods** of the parent class if you don’t change them inside and only call them with `super`. The parent methods are available in any inheritor of the class.